### PR TITLE
framework/st_things : Removing hash based device managing Logic

### PR DESCRIPTION
**Pull request by JeongJunSik**
… a hash map

- Logic changes to support only a single device because there are currently no multiple device requirements
- Define internal resource (easysetup, fota, etc.) as const str and change it to include in existing logic: Improvement of development convenience / readability

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [ff6ad4567416b31c05454d4ec746b5feb4198035] -

**Comment by seinfra** [comment]
Target : [ff6ad4567416b31c05454d4ec746b5feb4198035] - Code Rule Check Result:
framework/src/st_things/things_stack/framework/things_data_manager.c:1008: ERROR: [SPC_M_SEP] space required after that close brace '}'
framework/src/st_things/things_stack/framework/things_data_manager.c:1025: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
framework/src/st_things/things_stack/framework/things_data_manager.c:1035: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_data_manager.c:2033: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('

**Comment by seinfra** [comment]
Target : [ff6ad4567416b31c05454d4ec746b5feb4198035] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c541f01c0ac63cc37131e33f5d52b2c1417d45b5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c541f01c0ac63cc37131e33f5d52b2c1417d45b5] - Code Rule Check Result:
framework/src/st_things/things_stack/framework/things_data_manager.c:984: ERROR: [SPC_M_SEP] space required after that close brace '}'
framework/src/st_things/things_stack/framework/things_data_manager.c:1001: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
framework/src/st_things/things_stack/framework/things_data_manager.c:1011: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_data_manager.c:2009: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('

**Comment by seinfra** [comment]
Target : [c541f01c0ac63cc37131e33f5d52b2c1417d45b5] -

File: framework/src/st_things/things_stack/things_def.h [hunk 1]
```diff
@@ -51,4 +51,8 @@
 #define URI_DEVICE_COL                  "/device"
 #define URI_FIRMWARE                    "/firmware"
 
+#define DEVICE_OS_VERSION               "1.0"
```
Comment (Hyunjun85) [inline framework/src/st_things/things_stack/things_def.h]: Is it used?

Comment (JeongJunSik) [reply framework/src/st_things/things_stack/things_def.h]: Yes this is necessary.
I do not think user needs to enter this item.

**Comment by seinfra** [comment]
Target : [26de410c9398a994f63584c71a96a1aea9bb55c5] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [26de410c9398a994f63584c71a96a1aea9bb55c5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [26de410c9398a994f63584c71a96a1aea9bb55c5] -

**Comment by seinfra** [comment]
Target : [3b7e29e6846c3a35e8cd4ed3bb71d413d28fa5ea] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3b7e29e6846c3a35e8cd4ed3bb71d413d28fa5ea] - Code Rule Check OK.

File: framework/src/st_things/things_stack/framework/things_data_manager.c [hunk 2]
```diff
@@ -754,395 +714,240 @@ int parse_things_cloud_json(const char *filename)
...
+
+						if (setup_id != NULL) {
+							if (strlen(setup_id->valuestring) != 3) {
+								THINGS_LOG_V(TAG, "setup_id exceeds 3 bytes. please check (3 bytes are fixed sizes.)");
+								goto JSON_ERROR;
+							}
+							is_artik = false;
+							if (artik->type == cJSON_True) {
+								is_artik = true;
+							}
```
Comment (juitem) [inline framework/src/st_things/things_stack/framework/things_data_manager.c]: Logic is fine.
Reducing the complexity of nested conditions are recommended, How about changing the sequence of condition and reduce the depth of condition?

**Comment by seinfra** [comment]
Target : [4b7a62dc454dd433dfa3cec1054fdc5635de29a7] -

**Comment by seinfra** [comment]
Target : [4b7a62dc454dd433dfa3cec1054fdc5635de29a7] -

**Comment by seinfra** [comment]
Target : [75a3dc2be03f6e9b9a86800f613aa21cf31eb226] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [75a3dc2be03f6e9b9a86800f613aa21cf31eb226] - Code Rule Check Result:
framework/src/st_things/things_stack/framework/things_data_manager.c:755: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
framework/src/st_things/things_stack/framework/things_data_manager.c:759: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'

**Comment by seinfra** [comment]
Target : [2716be5125eede55c2fac93e0d4945db1702bb7f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2716be5125eede55c2fac93e0d4945db1702bb7f] - Code Rule Check OK.

**Pull request merged**